plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/FacetUtil.java

```diff
@@ -130,15 +142,32 @@ public FacetUtil addWebFacetToBatch(IProjectFacetVersion webFacet) {
       webAppDirectory = webAppDirectoryFound.toString();
     }
 
+    IDataModel webModel = createWebFacetDataModel(webAppDirectory);
+    facetInstallSet
+        .add(new IFacetedProject.Action(IFacetedProject.Action.Type.INSTALL, webFacet, webModel));
+    return this;
+  }
+
+  /**
+   * Create a Dynamic Web Facet configuration. The {@code webappFolder} is the location that holds
+   * the {@code WEB-INF/web.xml}.
+   */
+  public static IDataModel createWebFacetDataModel(IContainer webappFolder) {
+    return createWebFacetDataModel(webappFolder.getProjectRelativePath().toString());
+  }
+
+  /**
+   * Create a Dynamic Web Facet configuration.
+   */
+  public static IDataModel createWebFacetDataModel(String webInfFolder) {
```

Comment:
I think this is `webAppFolder`, not `webInfFolder`. (Also we should capitalize `App`.)